src/test/java/mate/academy/spring/boot/controllerTests/BookControllerTest.java

```diff
@@ -0,0 +1,289 @@
+        expected.add(secondBook);
+        MvcResult result = mockMvc.perform(get("/books"))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, BookDto.class));
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.isEmpty());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("Should return book by ID")
+    @WithMockUser
+    public void testGetFindById_withCorrectId_shouldReturnBook() throws Exception {
+        BookDto expected = getBookDto("Book 1", "Author 1", BigDecimal.valueOf(20.00));
+        Long bookId = 1L;
+
+        MvcResult result = mockMvc.perform(get("/books/{id}", bookId))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+        System.out.println();
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Should return EntityNotFoundException")
+    public void testGetBookById_withIncorrectId_ShouldReturnNotFound() throws Exception {
+        Long invalidBookId = 999L;
+        mockMvc.perform(get("/books/{id}", invalidBookId))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @DisplayName("Should search books by title")
+    @WithMockUser
+    public void getBooksByTitle_withCorrectTitle_shouldReturnListOfBooks() throws Exception {
+        List<BookDto> expected = new ArrayList<>();
+        BookDto firstBook = getBookDto("Book 1", "Author 1", BigDecimal.valueOf(20.00));
+        expected.add(firstBook);
+        String title = "Book 1";
+        MvcResult result = mockMvc.perform(get("/books/by-title")
+                        .param("title", title))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, BookDto.class));
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.isEmpty());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("Searching books by InvalidTitle")
+    @WithMockUser
+    public void getBooksByTitle_ShouldReturnEmpty_WhenNoBooksFound() throws Exception {
+        String nonExistentTitle = "Nonexistent Title";
+        mockMvc.perform(get("/books/by-title")
+                .param("title", nonExistentTitle))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()").value(0));
```

Comment:
I think it's better to use `objectMapper` to convert json to the `List<BookDto>` and then check the size. In my opinion it is easier to understand and debug, but not mandatory)